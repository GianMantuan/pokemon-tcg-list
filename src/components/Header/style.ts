import styled from 'styled-components/native';

export const HeaderContainer = styled.View`
  margin: 20px;
`;

export const Title = styled.Text`
  font-weight: bold
  font-size: 24px
  color: #121212
  margin-bottom: 10px  
`;

export const Subtitle = styled.Text`
font-weight: 400
font-size: 18px
color: #c1c7c9
`;
