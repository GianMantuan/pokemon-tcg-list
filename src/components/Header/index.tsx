import React from 'react';

import {HeaderContainer, Title, Subtitle} from './style';

type Props = {
  title: string;
  subtitle?: string;
};

function Header({title, subtitle}: Props): JSX.Element {
  return (
    <HeaderContainer>
      <Title>{title}</Title>
      {subtitle ? <Subtitle>{subtitle}</Subtitle> : null}
    </HeaderContainer>
  );
}

Header.defaultProps = {
  subtitle: false,
};

export default Header;
