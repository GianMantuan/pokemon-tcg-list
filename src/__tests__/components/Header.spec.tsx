import React from 'react'
import { Text } from 'react-native'
import Header from '../../components/Header'

import { render, cleanup } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

describe('Header Component', () => {
  afterEach(cleanup)

  it('should render Header component', () => {
    expect(renderer.create(<Header title="title" />)).toMatchSnapshot()
  })

  it('should not have subtitle rendered as a child', () => {
    const header = renderer.create(<Header title="title" />)

    expect(header.root.props.subtitle).toBeFalsy()
  })

  it('should have subtitle rendered as a child', () => {
    const header = renderer.create(<Header title="title" subtitle="subtitle" />)
    const loaderInst = header.root.findAllByType(Text)

    expect(loaderInst[1].props.children).toBe('subtitle')
  })
})