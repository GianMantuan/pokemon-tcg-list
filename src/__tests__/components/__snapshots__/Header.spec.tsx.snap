// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header component 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 20,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#121212",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 10,
        },
      ]
    }
  >
    title
  </Text>
</View>
`;
